spinalcordtoolbox/image.py

```diff
@@ -1603,3 +1603,33 @@ def _align_dict(dictionary, use_tabs=True, delimiter=""):
             padding = " " * (len_max - len(k))
         out.append(f"{k}{padding}{delimiter}{v}")
     return '\n'.join(out)
+
+
+def apply_mask_if_soft(fname_im: str, fname_mask: str):
+    """
+    Check to see if a mask is soft (non-binary), and if it is:
+        * Apply the mask directly to the image and save it as a separate image file.
+        * Set the mask filename to None to avoid the mask being used.
```

Comment:
I'm not sure if I love this behavior... it makes sense when you read the docstring, but it's not very clear if you're looking at just a function call:

```python
fname_im, fname_mask = image.apply_mask_if_soft(fname_im, fname_mask)
```

What happens to `fname_mask` here? You wouldn't know unless you checked.

I could instead do something like:

```python
fname_im, mask_was_applied = image.apply_mask_if_soft(fname_im, fname_mask)
if mask_was_applied:
    fname_mask = None
```

That seems a little more explicit...